lib/content-negotiation.js

```diff
@@ -43,6 +46,12 @@ function negotiateContentProducer(req, logWarning, options) {
     }
   }
 
+  // resolve renderer functions
+  sendHtml = require(options.htmlRenderer || './send-html');
```

Comment:
Overriding `sendHtml` that's shared in a global scope can create subtle bugs when `negotiateContentProducer` is called multiple times in parallel, each time with different `options` object.

I am proposing a different approach here:

 1. Instead of storing renderer functions in variables, store them in the `options` object itself.
 2. Modify `resolveOperation` to accept the `options` object, so that it can read renderer functions from there.
 3. Modifying properties of an object passed in as an argument is considered as anti-pattern, it often surprises users. Please shallow-clone the options object before making any changes: 

    ```js
    options = Object.assign({}, options);
    ```

Reply:
Very good. I'll combine this with the note above about top-level requires and figure it out.